### ISSUE TITLE
`cml pr`: Add `--merge` option.

### ISSUE THREAD

**Issue by daavoo** [iterative/cml#823]
It would be nice to have native support in CML for the cloud-agnostic equivalent of:

```
gh pr merge --rebase $(cml pr "foo")
```

**Comment by 0x2b3bfa0** [comment]
```diff
on: pull_request
jobs:
  cml:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: iterative/setup-cml@v1
      - name: Generate data
        run: echo "Hello World" > output.txt
      - name: Create and merge a new pull request
-        run: gh pr merge --rebase $(cml pr "output.txt")
+        run: cml pr --merge output.txt
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by daavoo** [comment]
> ```diff
> on: pull_request
> jobs:
>   cml:
>     runs-on: ubuntu-latest
>     steps:
>       - uses: actions/checkout@v2
>       - uses: iterative/setup-cml@v1
>       - name: Generate data
>         run: echo "Hello World" > output.txt
>       - name: Create and merge a new pull request
> -        run: gh pr merge --rebase $(cml pr "output.txt")
> +        run: cml pr --merge output.txt
>         env:
>           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
> ```

Things to consider, when using `gh` you have 1 level for `merge` and another for configuring how the merge is done `--rebase`. How would be the different merge options exposed in CML?

**Comment by daavoo** [comment]
> Probably cml pr --merge for merging, cml pr --rebase-merge / --merge-rebase / --rebase for rebasing and something similar for squashing if we choose to.

I kind of like `--merge`, `--rebase`, and `--squash`.

Maybe off-topic but, would it make any sense to have separate command `cml merge` ? 

So: `cml merge --rebase $(cml pr "output.txt")`?

**Comment by 0x2b3bfa0** [comment]
Yes, in an ideal world (#762) what is now known as `cml pr` should be `cml pr create` and it would make sense to have even both `cml pr merge` and `cml pr create --merge`

**Comment by casperdcl** [comment]
love `cml {pr,mr} {create.merge}`... updated #762 accordingly. Probably worth collecting similar breaking API changes there for a possible major version bump.

**Comment by casperdcl** [comment]
also how to handle conflicts? Avoid them à la [SO/70205896#3896283](stackoverflow.com/a/70205896/3896283)?

**Comment by 0x2b3bfa0** [comment]
See also automatic merge as in [`gh pr merge --auto`](https://cli.github.com/manual/gh_pr_merge) and https://github.com/iterative/cml/pull/903#issuecomment-1054362158

**Comment by 0x2b3bfa0** [comment]
Closed with #932; naming suggestions moved to https://github.com/iterative/cml/issues/762#issue-1024172986
